pkg/plugins/manager/loader/assetpath/assetpath.go

```diff
@@ -30,7 +30,11 @@ func DefaultService(cfg *config.Cfg) *Service {
 // Base returns the base path for the specified plugin.
 func (s *Service) Base(pluginJSON plugins.JSONData, class plugins.Class, pluginDir string) (string, error) {
 	if class == plugins.ClassCore {
-		return path.Join("public/app/plugins", string(pluginJSON.Type), filepath.Base(pluginDir)), nil
+		baseDir := getBaseDir(pluginDir, true)
+		if isDecoupledPlugin(pluginDir) {
+			return path.Join("public/plugins", baseDir), nil
```

Comment:
Just a note that hopefully long term we can do:
```suggestion
			return path.Join("public/plugins", pluginJSON.ID), nil
```

And make everything (including old core paths) follow that pattern, as right now it can be confusing.

Reply:
yep, right now it's not enforced and some plugins don't follow that convention. It's definitely something we can follow for decoupled plugins.